lib/pure/strmisc.nim

```diff
@@ -85,6 +86,41 @@ proc rpartition*(s: string, sep: string): (string, string, string)
 
   return partition(s, sep, right = true)
 
+func parseFloatThousandSep*(s: string; thousandSep = ','; floatSep = '.'): float {.since: (1, 3).} =
+  ## Convenience func for `parseFloat` but it can take 2 `char` separators,
+  ## one is **likely** the thousands separator, other is **likely** the floating point,
+  ## but if both separators are swapped it can still parse correctly, see the examples,
+  ## this is designed to parse floats as found in the wild formatted for humans.
+  ##
+  ## See also:
+  ## * `strutils <strutils.html>`_
+  runnableExamples:
+    doAssert parseFloatThousandSep("0") == 0.0
+    doAssert parseFloatThousandSep("0.0") == 0.0
+    doAssert parseFloatThousandSep("1.0") == 1.0
+    doAssert parseFloatThousandSep("-1.0") == -1.0
+    doAssert parseFloatThousandSep("1.000") == 1.0
+    doAssert parseFloatThousandSep("1,000") == 1.0
+    doAssert parseFloatThousandSep("10,000.000") == 10000.0
+    doAssert parseFloatThousandSep("10.000,000") == 10.0
+    doAssert parseFloatThousandSep("1,000,000.000") == 1000000.0
+    doAssert parseFloatThousandSep("1.000.000,000") == 1000000.0
+    doAssert parseFloatThousandSep("0000000000000001.00,0,000,00000,,00") == 1.0
+    doAssert parseFloatThousandSep("0000000000000001,00.0.000.00000..00") == 1.0
+    doAssert parseFloatThousandSep("000,1.000,,,,,,,,,,,,,,000,,,,,0000") == 1.0
```

Comment:
er... what? Since it performs some guesswork already it should be stricter and don't allow ",,," at all.

Reply:
Maybe I exaggerated the example too much to try to break it lol :laughing: